src/repr/src/proto/mod.rs

```diff
@@ -140,6 +142,23 @@ impl ProtoRepr for u8 {
     }
 }
 
+impl ProtoRepr for u128 {
+    type Repr = ProtoU128;
+
+    fn into_proto(self: Self) -> Self::Repr {
+        let lo = (self & (u64::MAX as u128)) as u64;
+        let hi = self >> 64;
+        ProtoU128 {
+            hi: hi.try_into().unwrap(),
+            lo,
```

Comment:
Why not simply 
```rust
let hi = (self >> 64) as u64;
```
here?

Reply:
Ah it compiles! Good catch!

I originally tried
```
let hi = self >> 64 as u64;
```
error[E0308]: mismatched types
but the compiler told me
```
   --> src/repr/src/proto/mod.rs:152:13
    |
152 |             hi,
    |             ^^ expected `u64`, found `u128`
    |
help: you can convert a `u128` to a `u64` and panic if the converted value doesn't fit
    |
152 |             hi: hi.try_into().unwrap(),
    |             +++   ++++++++++++++++++++
```
so I followed the compiler's advice and did not realize my mistake was that only the `64` was being converted to `u64`.